extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaProcessor.java

```diff
@@ -141,6 +141,9 @@ public void build(CombinedIndexBuildItem indexBuildItem, BuildProducer<Reflectiv
                     new ReflectiveClassBuildItem(false, false, ObjectMapperSerializer.class, ObjectMapperDeserializer.class));
             collectSubclasses(toRegister, indexBuildItem, ObjectMapperSerializer.class);
             collectSubclasses(toRegister, indexBuildItem, ObjectMapperDeserializer.class);
+
+            // Make the `io.quarkus.jackson.runtime.ObjectMapperProducer` bean cannot be removed.
+            beans.produce(UnremovableBeanBuildItem.beanClassNames("com.fasterxml.jackson.databind.ObjectMapper"));
```

Comment:
I don't think this would work because `Bean#getBeanClass()` returns the class that declares the producer method, `io.quarkus.jackson.runtime.ObjectMapperProducer` for [this producer method](https://github.com/quarkusio/quarkus/blob/master/extensions/jackson/runtime/src/main/java/io/quarkus/jackson/runtime/ObjectMapperProducer.java#L24) . It should be `UnremovableBeanBuildItem.beanTypes("com.fasterxml.jackson.databind.ObjectMapper")` instead.

Reply:
you're right, I changed it. 

Any reason by `beanTypes` does not accept String directly?

Reply:
Because `Class` and `DotName` were enough until now? ;-)